consensus/service/src/tx_manager.rs

```diff
@@ -351,43 +334,387 @@ impl<E: ConsensusEnclaveProxy, L: Ledger, UI: UntrustedInterfaces> TxManager<E,
         Ok(self.enclave.txs_for_peer(&encrypted_txs?, aad, peer)?)
     }
 
-    pub fn get_encrypted_tx_by_hash(&self, tx_hash: &TxHash) -> Option<WellFormedEncryptedTx> {
-        self.lock_cache()
+    /// Get the encrypted transaction corresponding to the given hash.
+    fn get_encrypted_tx(&self, tx_hash: &TxHash) -> Option<WellFormedEncryptedTx> {
+        self.well_formed_cache
             .get(tx_hash)
             .map(|entry| entry.encrypted_tx().clone())
     }
-
-    pub fn num_entries(&self) -> usize {
-        self.lock_cache().len()
-    }
-
-    fn lock_cache(&self) -> MutexGuard<HashMap<TxHash, CacheEntry>> {
-        self.cache.lock().expect("lock poisoned")
-    }
 }
 
 #[cfg(test)]
-mod tests {
+mod tx_manager_tests {
     use super::*;
     use crate::validators::DefaultTxManagerUntrustedInterfaces;
-    use mc_common::logger::test_with_logger;
+    use mc_attest_core::{IasNonce, Quote, QuoteNonce, Report, TargetInfo, VerificationReport};
+    use mc_attest_enclave_api::{
+        ClientAuthRequest, ClientAuthResponse, ClientSession, EnclaveMessage, PeerAuthRequest,
+        PeerAuthResponse, PeerSession,
+    };
+    use mc_common::{logger::test_with_logger, ResponderId};
+    use mc_consensus_enclave::{Error as EnclaveError, LocallyEncryptedTx, SealedBlockSigningKey};
     use mc_consensus_enclave_mock::ConsensusServiceMockEnclave;
+    use mc_crypto_keys::{Ed25519Public, X25519Public};
+    use mc_ledger_db::Ledger;
+    use mc_sgx_report_cache_api::{Error as SgxReportError, ReportableEnclave};
     use mc_transaction_core_test_utils::{
         create_ledger, create_transaction, initialize_ledger, AccountKey,
     };
     use rand::{rngs::StdRng, SeedableRng};
 
+    // ConsensusEnclave inherits from ReportableEnclave, so the traits have to be re-typed here.
+    // Splitting the traits apart might help because TxManager only uses a few of these functions.
+    mock! {
+        Enclave {}
```

Comment:
Seems like this should live somewhere else since I can imagine it will get re-used wherever a consensus enclave needs to be mocked?

Reply:
I agree. I think this is the first of several PRs I have in mind that will use this, so I'll probably end up moving it soon, maybe into consensus/enclave/api, where the ConsensusEnclave trait is defined.